homeassistant/components/switchbot/config_flow.py

```diff
@@ -84,36 +71,59 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return SwitchbotOptionsFlowHandler(config_entry)
 
+    def __init__(self):
+        """Initialize the config flow."""
+        self._discovered_devices = {}
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initiated by the user."""
 
-        errors = {}
+        errors: dict[str, str] = {}
 
         if user_input is not None:
-            user_input[CONF_MAC] = user_input[CONF_MAC].replace("-", ":").lower()
+            await self.async_set_unique_id(user_input[CONF_MAC].replace(":", ""))
+            self._abort_if_unique_id_configured()
+
+            user_input[CONF_SENSOR_TYPE] = SUPPORTED_MODEL_TYPES[
+                self._discovered_devices[self.unique_id]["modelName"]
+            ]
 
-            # abort if already configured.
-            for item in self._async_current_entries():
-                if item.data.get(CONF_MAC) == user_input[CONF_MAC]:
-                    return self.async_abort(reason="already_configured_device")
+            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
 
-            try:
-                return await self._validate_mac(user_input)
+        try:
+            self._discovered_devices = await self._get_switchbots()
 
-            except NotConnectedError:
-                errors["base"] = "cannot_connect"
+        except NotConnectedError:
+            return self.async_abort(reason="cannot_connect")
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            return self.async_abort(reason="unknown")
+
+        # Get devices already configured.
+        configured_devices = {
+            item.data[CONF_MAC]
+            for item in self._async_current_entries(include_ignore=False)
+        }
+
+        # Get supported devices not yet configured.
+        unconfigured_devices = {
+            device["mac_address"]
```

Comment:
If you make this a dict you can show the modelName and the MAC address as the value (probably construct with a. F-string). It might make it a bit easier for the user to figure out which device is which if they have multiple.

Reply:
<img width="382" alt="Screen Shot 2021-09-27 at 2 38 59 PM" src="https://user-images.githubusercontent.com/663432/134973778-5eaa0ca7-0c7a-4701-9413-049aea19b6a1.png">

I went ahead and took care of this as it would be nice to have the device selection in beta.